content/docs/buildpack-basics.md

```diff
@@ -0,0 +1,63 @@
++++
+title="Basics of BuildPacks"
+weight=2
+getting-started=true
++++
+
+## Basic Concepts
+
+### What is a Buildpack?
+
+A `buildpack` is software that transforms application source code into
+executable files by analyzing the code and determining the best way to
+build it.
```

Comment:
I've always found it difficult to describe what an individual buildpack is!  A `buildpack` is a small software component that does something.  Would we be better off here to explain by example?  i.e. provide an example of a "distribution" buildpack, an a "build process" buildpack?

Reply:
- How about adding the following paragraph under this definition:

_Buildpacks have many shapes and forms. For example, a 'Distribution BuildPack' is a pre-built and tested buildpack that is ready for distribution to application developers. A Distribution Buildpack includes a set of buildpacks that are packaged together and can be used to build applications in different environments._

_'Paketo Buildpacks' is a distribution buildpack for building applications in Java, Go, Python, Ruby, etc. The buildpacks included in 'Paketo Buildpacks' work together to create application container images that can run on any platform that supports container images (e.g., Cloud Foundry, Docker, Kubernetes, etc.)._ 

Source: https://paketo.io/

- I couldn't find much information on "build process" buildpacks, can you please elaborate on this?

P.s., I'm still connecting the dots with the core concepts, please see if the description is conceptually correct.